cli/deployment/deployment.go

```diff
@@ -0,0 +1,78 @@
+package deployment
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+)
+
+// Deployment object represent real GCP Deployment entity that already created or have to be created
+// configFile field point to yaml file, generated from Config object with all cross-deployment references
+// overwritten with actual values
+type Deployment struct {
+	Outputs    map[string]string
+	config     Config
+	configFile string
+}
+
+// NewDeployment creates Deployment object and override all out refs, this means all
+// deployments it depends to should exists in GCP project
```

Comment:
"NewDeployment creates a new Deployment object, overriding all outward references. In effect, this means all deployment dependencies must exist in the GCP project."